### PR TITLE
Restart message when a java path preference is changed

### LINKED ISSUE #150: JDK changes (xml.java.home or java.home) should be detected
... and user should be prompted to restart vscode (similar to xml.server.vmargs and vscode-java)

### PR DESCRIPTION
**Pull request by NikolasKomonen**
Fixes #150

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    }
+
+    const newConfig = getXMLConfiguration();
+
+    if(hasPreferenceChanged(oldConfig, newConfig, "java.home")) { // checks "xml.java.home"
+      const message = "`xml.java.home` path has changed. Please restart VS Code."
+      createReloadWindowMessage(message);
+    }
+
+    else if(javaHomeConfig != getJavaHomeConfiguration()) { // checks "java.home"
```
Comment (fbricon) [inline src/settings.ts]: on check it if xml.java.home is not set, else you can ignore the change, it doesn't impact XML

**Comment by fbricon** [comment]
the change should be refactored along with verifyVMArgs. Now we have 2 change listeners

**Comment by NikolasKomonen** [comment]
@fbricon updated, moved into 1 listener in the settings.ts file

File: src/settings.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+export function onConfigurationChange() {
+  if(!ignoreAutoCloseTags) {
+    verifyAutoClosing();
+  }
+  
+  !ignoreVMArgs ? verifyVMArgs(): undefined;
+
+  workspace.onDidChangeConfiguration(params => {
```
Comment (fbricon) [inline src/settings.ts]: this seems wrong. This needs to be registered once. Calling it from inside middleware.workspace.didChangeConfiguration doesn't seem right. There must be a way to get the change event directly from middleware.workspace.didChangeConfiguration

**Comment by fbricon** [comment]
@snjeza do you mind looking into this one, while @NikolasKomonen is away?

File: src/settings.ts [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+export function getXMLConfiguration(): WorkspaceConfiguration {
+  return workspace.getConfiguration("xml");
+}
+
+export function onConfigurationChange() {
+  if(!ignoreAutoCloseTags) {
+    verifyAutoClosing();
+  }
+  
+  !ignoreVMArgs ? verifyVMArgs(): undefined;
```
Comment (fbricon) [inline src/settings.ts]:
```typescript
  if (!ignoreVMArgs) {
    verifyVMArgs();
  }
```

**Comment by NikolasKomonen** [comment]
@fbricon Updated, setup the jdk listener to subscribe only once

It starts at this line: https://github.com/redhat-developer/vscode-xml/blob/a190861729b680bc04da2e67cef58e823a608358/src/extension.ts#L94

**Comment by NikolasKomonen** [comment]
@fbricon 

Updated,

The prompt should appear when the 'xml.java.home' setting is changed.
If 'java.home' is changed but 'xml.java.home' exists, then nothing happens.
If 'java.home' changes, 'xml.java.home' doesn't exist, and the Java extension is inactive/missing, then you get the prompt.

**Pull request merged**